crates/ironrdp-pdu/src/gcc/network_data.rs

```diff
@@ -12,17 +12,92 @@ const CHANNELS_MAX: usize = 31;
 
 const CLIENT_CHANNEL_COUNT_SIZE: usize = 4;
 
-const CLIENT_CHANNEL_NAME_SIZE: usize = 8;
 const CLIENT_CHANNEL_OPTIONS_SIZE: usize = 4;
-const CLIENT_CHANNEL_SIZE: usize = CLIENT_CHANNEL_NAME_SIZE + CLIENT_CHANNEL_OPTIONS_SIZE;
+const CLIENT_CHANNEL_SIZE: usize = ChannelName::SIZE + CLIENT_CHANNEL_OPTIONS_SIZE;
 
 const SERVER_IO_CHANNEL_SIZE: usize = 2;
 const SERVER_CHANNEL_COUNT_SIZE: usize = 2;
 const SERVER_CHANNEL_SIZE: usize = 2;
 
+/// An 8-byte array containing a null-terminated collection of seven ANSI characters
+/// with the purpose of uniquely identifying a channel.
+///
+/// In RDP, an ANSI character is a 8-bit Windows-1252 character set unit. ANSI character set
+/// is using all the code values from 0 to 255, as such any u8 value is a valid ANSI character.
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ChannelName {
+    inner: Cow<'static, [u8; Self::SIZE]>,
+}
+
+impl ChannelName {
+    pub const SIZE: usize = 8;
+
+    /// Creates a channel name using the provided array, ensuring the last byte is always the null terminator.
+    pub const fn new(mut value: [u8; Self::SIZE]) -> Self {
+        value[Self::SIZE - 1] = 0; // ensure the last byte is always the null terminator
+
+        Self {
+            inner: Cow::Owned(value),
+        }
+    }
+
+    /// Converts an UTF-8 string into a channel name by copying up to 7 bytes.
+    pub fn from_utf8(value: &str) -> Option<Self> {
+        let mut inner = [0; Self::SIZE];
+
+        value
+            .chars()
+            .take(Self::SIZE - 1)
+            .zip(inner.iter_mut())
+            .try_for_each(|(src, dst)| {
+                let c = u8::try_from(src).ok()?;
+                c.is_ascii().then(|| *dst = c)
+            })?;
+
+        Some(Self {
+            inner: Cow::Owned(inner),
+        })
+    }
```

Comment:
This would be more useful if it just returned a `Self` and panicked on any errors. With `Option<Self>` it makes it impossible to use like

```rust
const CHANNEL_NAME: ChannelName = ChannelName::from_utf8("cliprdr");
```

Reply:
You won’t be able to use it like this because `from_utf8` is not `const`-friendly (and can’t be made `const` because we can’t use iterators in const context as of now). That’s why `from_utf8` is only a convenience function for dynamic contexts where it’s preferable to catch the error instead of panicking, and that’s why we also have `ChannelName::from_static`.

Reply:
(Though it would be nice to revisit this when `const` is made more powerful in some future Rust version)